converters/provider-aws/common.go

```diff
@@ -0,0 +1,227 @@
+	"github.com/upbound/extensions-migration/converters/provider-aws/efs"
+	"github.com/upbound/extensions-migration/converters/provider-aws/eks"
+	"github.com/upbound/extensions-migration/converters/provider-aws/elasticache"
+	"github.com/upbound/extensions-migration/converters/provider-aws/iam"
+	"github.com/upbound/extensions-migration/converters/provider-aws/kms"
+	"github.com/upbound/extensions-migration/converters/provider-aws/rds"
+	"github.com/upbound/extensions-migration/converters/provider-aws/route53"
+	"github.com/upbound/extensions-migration/converters/provider-aws/route53resolver"
+	"github.com/upbound/extensions-migration/converters/provider-aws/s3"
+	"github.com/upbound/extensions-migration/converters/provider-aws/secretsmanager"
+	"github.com/upbound/extensions-migration/converters/provider-aws/sns"
+	"github.com/upbound/extensions-migration/converters/provider-aws/sqs"
+)
+
+// DefaultPatchSetsConverter is a default patchset converter for provider-aws
+func DefaultPatchSetsConverter(sourcePatchSets map[string]*xpv1.PatchSet) error {
+	tagsPatchSetName := ""
+	for _, patchSet := range sourcePatchSets {
+		for _, patch := range patchSet.Patches {
+			if patch.ToFieldPath != nil {
+				if strings.HasPrefix(*patch.ToFieldPath, "spec.forProvider.tags") {
+					tagsPatchSetName = patchSet.Name
+					break
+				}
+			}
+		}
+		if tagsPatchSetName != "" {
+			break
+		}
+	}
+
+	tPs := sourcePatchSets[tagsPatchSetName]
+	if tPs == nil {
+		return nil
+	}
+	for i, p := range tPs.Patches {
+		r := strings.NewReplacer("metadata.labels[", "", "]", "")
+		key := r.Replace(*p.FromFieldPath)
+		*tPs.Patches[i].ToFieldPath = fmt.Sprintf(`spec.forProvider.tags[%s]`, key)
+	}
+	// convert patch sets in the source
+	return nil
+}
+
+// ConvertComposedTemplateTags is responsible converting the tags of provider-aws resources
+func ConvertComposedTemplateTags(sourceTemplate xpv1.ComposedTemplate) ([]xpv1.Patch, error) {
+	var patchesToAdd []xpv1.Patch
+	for _, p := range sourceTemplate.Patches {
+		if p.ToFieldPath != nil {
+			if strings.HasPrefix(*p.ToFieldPath, "spec.forProvider.tags") {
+				u, err := migration.FromRawExtension(sourceTemplate.Base)
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to convert ComposedTemplate")
+				}
+				paved := fieldpath.Pave(u.Object)
+				key, err := paved.GetString(strings.ReplaceAll(*p.ToFieldPath, ".value", ".key"))
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to get value from paved")
+				}
+				s := fmt.Sprintf(`spec.forProvider.tags["%s"]`, key)
+				patchesToAdd = append(patchesToAdd, xpv1.Patch{
+					FromFieldPath: p.FromFieldPath,
+					ToFieldPath:   &s,
+					Transforms:    p.Transforms,
+					Policy:        p.Policy,
+				})
+			}
+		}
+	}
+	return patchesToAdd, nil
+}
+
+// RegisterAllKnownConverters registers all known converters for provider-aws
+func RegisterAllKnownConverters(r *migration.Registry) {
```

Comment:
Please add a note to this function to remind to register any feature API converters implemented for the community AWS provider.

We may also discuss this in the package documentation.

Reply:
Added a note!